docs/ds20.md

```diff
@@ -0,0 +1,112 @@
+Clearly this isn't good when the majority of firmware updates are distributed to address security issues.
+
+One suggestion would be to put this quirk information into the existing update metadata provided by the configured remote, but this has several problems:
+
+* The daemon needs to *enumerate* hardware that does not have updates on the main LVFS remote.
+* A *lot* of machines using fwupd have never connected to the internet and we still want to enumerate hardware for audit and verification purposes.
+* Re-enumerating all physical hardware (to get the latest quirks) when updating the metadata is not straightforward.
+* The VID/PID is not necessarily the information that the plugin matches to assign the firmware stream, and so this would have to be a separate facet of metadata -- which may have to include quirks too.
+
+Matching devices to plugins should be thought of as *orthogonal* to matching firmware to devices.
+To simplify deployment it should be possible to allow the device itself to specify the plugin to use and optionally additional quirk data.
+
+## Specification
+
+Devices that implement a fwupd BOS DS20 descriptor can be updated without always needing to update the runtime version of fwupd for updated quirk entries, assuming the device is updatable by the existing vendor plugin.
+
+USB devices can create a new platform device capability BOS *“Binary Object Store”* descriptor with `bDevCapabilityType=0x05` and a fwupd-specific UUID, specifically `010aec63-f574-52cd-9dda-2852550d94f0`.
+This UUID is generated type-5 SHA-1 hash of the word `fwupd` using a DNS namespace.
+Using this custom UUID will ensure that Microsoft Windows does not try to parse the capability descriptor as a *Microsoft OS 2.0 descriptor set*.
+
+## Implementation
+
+Create a BOS DS20 descriptor such as:
+
+    1C 10 05 00 63 ec 0a 01 74 f5 cd 52 9d da 28 52 55 0d 94 f0 05 08 01 00 20 00 2a 00
+    ├┘ ├┘ ├┘ ├┘ └─────────────┬───────────────────────────────┘ └─┬───────┘ └─┬─┘ ├┘ ├┘
+    │  │  │  │                └─PlatformCapabilityUUID            │   wLength─┘   │  │
+    │  │  │  └─bReserved                                dwVersion─┘   bVendorCode─┘  │
+    │  │  └────bDevCapability                                         bAltEnumCmd────┘
+    │  └───────bDescriptorType
+    └──────────bLength
+
+The `dwVersion` encoded here is fwupd `1.8.5`, which is also the first version that supports BOS DS20 descriptors.
+
+The BOS descriptors are sorted by the requester and can appear in any order.
+The descriptor with the highest `dwVersion` that is not newer than the current running daemon version is used.
+This means that `dwVersion` is effectively the minimum version of fwupd that should read the descriptor.
+This allows devices to set different quirks depending on the fwupd version, although in practice this should not be required.
+
+* The `dwVersion` parameter **must not** be zero.
```

Comment:
how about must be 1.8.5 or newer (to make it clear that we don't support it until such a version)

Reply:
Fixed, ta.